### ISSUE TITLE
Question: How to use hinted-compilation to partially compile a project, but get a complete distribution?

### ISSUE THREAD

**Issue by math-artist** [Nuitka/Nuitka#563]
Is it possible to set hinted-compilation to create a standalone distribution, but set specific modules or scripts to be included in the distribution, but not compiled? 

Reasons: 
1. Quick work-around for compatibility issues 
2. Be LGPL compliant

With get-hints.py, we already get all the required modules for the distribution, so that part is done before compile. What would be awesome is an option to set a module to be included but not compiled. This could be especially useful for projects with many dependencies.

Is there already a workflow/work-around to do this?

**Comment by JorjMcKie** [comment]
What do you mean by "included but not compiled"?

I am not aware of a way to include modules as ``.py`` sources, and still have the compilation result be executable as a whole.
It is of course always posible to include arbitrary files as data files.

And you can exclude the compilation to C code on module level and include it as bytecode (equivalent to ``.pyc``), which makes it executable as before, but of course with an unchanged performance.
This technique is already in use for
* "unworthy" modules / packages (e.g. testing)
* special cases whose size would simply kill compile time (torch)
* there plugin interfaces available, which allow to switch of compilation to C and instead include any module as bytecode
* while technically this is possible, the question remains, how you want to specify this. For serious apps, including hundreds of modules is typical. I am sure, you don't want to maintain any config file that large for some or all of you apps.
* maybe a plugin option for certain large packages (numpy, matplotlib, ...) may do, or a special, general purpose plugin, which accepts a list of modules to keep as byte code ...

Could you elaborate a bit more about your idea based on the previous comments?
How this impact license situations - I don't know.

**Comment by math-artist** [comment]
> And you can exclude the compilation to C code on module level and include it as bytecode (equivalent to .pyc), which makes it executable as before, but of course with an unchanged performance.

This! A general purpose plug-in would be useful to granulary compile a few modules to .pyc. This would solve point 1.

To be LGPL compliant, a library must be linked to dynamically, the code needs to be outside of the executable. I was able to do that last week with PySide2 manually. I created a small main script that calls a PySide2 dependent project. I compiled the main as a standalone without recursing, and then I manually copied PySide2 and my module in the root, and it worked. 

Is there a more convenient way to achieve the equivalent with a plugin?
